src/app/adminDashboard/users/page.tsx

```diff
@@ -1,7 +1,120 @@
-import React from 'react';
+'use client';
+import React, { useState, useEffect } from 'react';
+import styles from '../../../styles/Dashboard.module.scss';
+import { Navbar } from '../../../Components/Navbar/Navbar';
+import TableData from '@/Components/CommonComponent/Table/Table';
+//@ts-ignore
+import { SearchFilter } from '@/Components/CommonComponent/SearchFilter';
+import RequireAuthentication from '../../../utils/requireAuthentication';
+// import apiServices from '@/services/api/apiServices';
+import { sendRequest } from '@/utils/axiosInstanse';
+import Loader from '@/Components/CommonComponent/Loader/Loader';
+import { toast } from 'react-toastify';
+import {SpectatorDataType} from '.././spectator/page';
 
-const page = () => {
-  return <div>Users</div>;
-};
+export interface IAppProps { }
 
-export default page;
+
+function page() {
+  const [ wholeUserData, setWholeUserData] = useState<SpectatorDataType[] | []>([]);
+  const [userData, setUserData] = useState<SpectatorDataType[] | []>([]);;
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+  const imageIcon: string = 'user';
+  const columns: string[] = [
+    'Full Name',
+    'User Name',
+    'Email',
+  ];
+
+  const fetchTournaments = async (searchVal:string) => {
+    try {
+      const token = localStorage.getItem('jwtToken');
+      const response = await sendRequest(`/user/getalluser?search=${searchVal}`, {
```

Comment:
is it possible to a create service for all these apis and use that service here? please check services folder

Reply:
resolved